### PR TITLE
feat: imporvements to pachyderm chart

### PR DESCRIPTION
**Pull request by mbelang**
- De-hardcode suite label to allow deployment in a different namespace
than `pachyderm`
- De-hardcode service account name to allow deployment in a different
namespace than `pachyderm`
- Allow specifying resource limits for `etcd` and `patchd` deployments
- Split `pachd` service in 3 stand alone services to allow exposing
`pachd-grpc-api` through a LoadBalancer service type
- `pachd-grpc-api` port can be customized
- `pachd` deployment now support iam-role annotation for `kube2iam`
- Fixed some linting issues in the Readme.md

#### What this PR does / why we need it:
The main feature of this PR is to be able to expose the grpc api throught an external loadbalancer.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:
Not that the changes are totally backward compatible but a `minor` version bump feels right b/c of the amount of improvements and changes herein.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by jonandernovella** [comment]
@mbelang First off, thanks for the well documented and explained PR! Have you been able to test it with the different storage backend options, including `hostPath`?

\Jon

**Comment by mbelang** [comment]
@jonandernovella You mean other that AWS? If I understand the question correctly no I didn't because we are on AWS only.

I did a test with default values which result in everything working fine as it was before the change (noop).

**Comment by jonandernovella** [comment]
@mbelang Right. Before we can merge this PR we need to make sure the rest of the deployments work. There was a change with regard to storage backend starting from v1.8. The manifesto of the`pachd` deployment now includes several new `env` vars.

E.g: The minio variables now needs to be retrieved from the secrret as follows:

```		{ "name": "MINIO_ENDPOINT", 
                  "valueFrom": {
                    "secretKeyRef": { 
                      "name": "pachyderm-storage-secret",
                      "key": "minio-endpoint",
                      "optional": true 
                      } 
                    }
                 }
 ```
\Jon

**Comment by mbelang** [comment]
Ok so it has to do with the `pachd` upgrade from 1.7.x to 1.8.x right? Not really with the rest of the changes IIRC?

**Comment by jonandernovella** [comment]
@mbelang Exactly. That upgrade was the one that introduced the changes.

**Comment by mbelang** [comment]
@jonandernovella, I'm currently not in the position to test that as I do not use `minIO`. Regarding what you describe doesn't seem to be supported just by taking a quick look at the templates.

**Comment by mbelang** [comment]
@jonandernovella instead of bumping the `pachyderm` version to `1.8.x`, I could revert to `1.7.x` so it wont break existing behavior and we could work on making support for 1.8.3 later. I my side I do not use `minio` so I can specify `1.8.x` version on my values.yaml.

File: stable/pachyderm/templates/pachd_deployment.yaml [hunk 1]
```diff
@@ -54,6 +58,10 @@ spec:
           value: GOOGLE
           {{- else if eq .Values.credentials "amazon" }}
           value: AMAZON
+        - name: AMAZON_REGION
```
Comment (jonandernovella) [inline stable/pachyderm/templates/pachd_deployment.yaml]: These changes need to be made regarding the env vars:
```
- name: GOOGLE_BUCKET
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: google-bucket
      optional: true
- name: GOOGLE_CRED
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: google-cred
      optional: true
- name: AMAZON_BUCKET
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: amazon-bucket
      optional: true
- name: AMAZON_DISTRIBUTION
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: amazon-distribution
      optional: true
- name: AMAZON_ID
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: amazon-id
      optional: true
- name: AMAZON_SECRET
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: amazon-secret
      optional: true
- name: AMAZON_REGION
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: amazon-region
      optional: true
- name: AMAZON_TOKEN
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: amazon-token
      optional: true
- name: MICROSOFT_CONTAINER
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: microsoft-container
      optional: true
- name: MICROSOFT_ID
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: microsoft-id
      optional: true
- name: MICROSOFT_SECRET
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: microsoft-secret
      optional: true
- name: MINIO_ENDPOINT
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: minio-endpoint
      optional: true
- name: MINIO_BUCKET
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: minio-bucket
      optional: true
- name: MINIO_SECURE
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: minio-secure
      optional: true
- name: MINIO_ID
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: minio-id
      optional: true
- name: MINIO_SECRET
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: minio-secret
      optional: true
- name: MINIO_SIGNATURE
  valueFrom:
    secretKeyRef:
      name: pachyderm-storage-secret
      key: minio-signature
      optional: true
```

**Review by jonandernovella — CHANGES_REQUESTED** [review]
Apart from those variables I wrote before, perhaps it's also nice to add one for exposing the S3 API, which would require a boolean variable.

```
- name: EXPOSE_OBJECT_API
  value: {{ .Values.exposeObjApi }}
```

**Comment by mbelang** [comment]
@jonandernovella that should do it :) sorry for the delay.

**Comment by jonandernovella** [comment]
@mbelang No worries. Good job!

**Comment by jonandernovella** [comment]
@mbelang The test seems to fail. The offending line might be: ``` I0513 07:48:53.880] Error: release pachyderm-6n2ucwluje failed: Service "pachd-api-grpc" is invalid: spec.ports[0].nodePort: Invalid value: 650: provided port is not in the valid range. The range of valid ports is 30000-32767```

File: stable/pachyderm/values.yaml [hunk 2]
```diff
@@ -41,22 +42,31 @@ microsoft:
...
   requests:
    cpu: 250m
    memory: 512M
+  # limits:
+  #  cpu: 250m
+  #  memory: 512M
+ service:
+  grpc:
+   annotations: {}
+   port: 650
```
Comment (jonandernovella) [inline stable/pachyderm/values.yaml]: We could default it to `30650`.

**Review by jonandernovella — CHANGES_REQUESTED** [review]
Plesse fix the issue wrt the port range so that it passes the test.

File: stable/pachyderm/values.yaml (hunk 2 above)
Comment (mbelang) [reply stable/pachyderm/values.yaml]: done, this is the default port, I just forgot to change the port I use for my loadBalancer type :).

**Review by jonandernovella — CHANGES_REQUESTED** [review]
New error in the test: ``` Error: release pachyderm-tj6h4f4azm failed: Deployment in version "v1beta1" cannot be handled as a Deployment: v1beta1.Deployment.Spec: v1beta1.DeploymentSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.Containers: []v1.Container: v1.Container.Env: []v1.EnvVar: v1.EnvVar.Value: ReadString: expects " or n, but found f, error found in #10 byte of ...|,"value":false},{"na|..., bigger context ...|rs":[{"env":[{"name":"EXPOSE_OBJECT_API","value":false},{"name":"PACH_ROOT","value":"/pach"},{"name"|...```

**Comment by jonandernovella** [comment]
@mbelang It seems that we are facing the same error again.

**Comment by mbelang** [comment]
Will do proper debugging locally then

**Comment by mbelang** [comment]
I hope this one is the last one haha :)

**Review by jonandernovella — APPROVED** [review]

**Pull request merged**